ModManagerUI/ModsBox.cs

```diff
@@ -0,0 +1,228 @@
+        private readonly IAddonService _addonService;
+        private readonly VisualElementInitializer _visualElementInitializer;
+        private readonly IResourceAssetLoader _resourceAssetLoader;
+        private VisualElement _mods;
+        private Filter _filter = ModFilter.Downloads.Desc();
+        private Label _loading;
+        private Label _error;
+        private TextField _search;
+        private RadioButtonGroup _tags;
+        private List<string> _tagOptions;
+        private ExtractorService _extractor;
+
+        private AssetBundle _bundle;
+
+        public ModsBox(VisualElementLoader visualElementLoader,
+                       PanelStack panelStack,
+                       IAddonService addonService,
+                       VisualElementInitializer visualElementInitializer,
+                       IResourceAssetLoader resourceAssetLoader,
+                       ExtractorService extractor)
+        {
+            _visualElementLoader = visualElementLoader;
+            _panelStack = panelStack;
+            _addonService = addonService;
+            OpenOptionsDelegate = OpenOptionsPanel;
+            _visualElementInitializer = visualElementInitializer;
+            _resourceAssetLoader = resourceAssetLoader;
+            _extractor = extractor;
+
+            _bundle = AssetBundle.LoadFromFile(@"D:\Ohjelmat\Steam\steamapps\common\Timberborn\BepInEx\plugins\ModManager\assets\what.bundle");
+            //_bundle = AssetBundle.LoadFromFile($"{Path.Combine(Paths.ModManager.Assets, "what.bundle")}");
+        }
+
+        public VisualElement GetPanel()
+        {
+            string assetName = "assets/resources/ui/views/mods/modsbox.uxml";
+            var asset = _bundle.LoadAsset<VisualTreeAsset>(assetName);
+            var root = LoadVisualElement(asset);
+
+            _mods = root.Q<ScrollView>("Mods");
+            _loading = root.Q<Label>("Loading");
+            _error = root.Q<Label>("Error");
+            _tags = root.Q<RadioButtonGroup>("Tags");
+
+            ShowModsAndTags();
+
+            root.Q<Button>("Close").clicked += OnUICancelled;
+            root.Q<Button>("SearchButton").clicked += UpdateMods;
+            _search = root.Q<TextField>("Search");
+            _search.isDelayed = true;
+            _search.RegisterValueChangedCallback(_ => UpdateMods());
+            return root;
+        }
+
+        private void OpenOptionsPanel()
+        {
+            _panelStack.HideAndPush(this);
+        }
+
+        public bool OnUIConfirmed()
+        {
+            return false;
+        }
+
+        public void OnUICancelled()
+        {
+            _panelStack.Pop(this);
+        }
+
+        private void ShowModsAndTags()
+        {
+            _loading.ToggleDisplayStyle(true);
+            _error.ToggleDisplayStyle(false);
+            _mods.Clear();
+
+            var getModsTask = _addonService.GetMods().Search(_filter).ToList();
+            getModsTask.ConfigureAwait(true).GetAwaiter()
```

Comment:
Maybe we need a helper for this to make it easier with like a callback methods

Reply:
🤔 maybe, I feel like something needs to be changed to it for sure. We'll wait for pmdudas UI additions